lib/std/date.zig

```diff
@@ -0,0 +1,15 @@
+pub const gregorian = @import("./date/gregorian.zig");
+pub const epoch = @import("./date/epoch.zig");
+
+pub const Date = gregorian.Date;
+pub fn DatePosix(comptime Year: type) type {
+    return gregorian.Date(Year, epoch.posix);
+}
+
+pub const Date16 = DatePosix(i16);
+pub const Date32 = DatePosix(i32);
+pub const Date64 = DatePosix(i64);
```

Comment:
Is there a particular reason these are generic? Shouldn't a single `Date` with `i32` as the year suffice?

Reply:
I made `Date` generic on `Year` to begin with because if you're storing many of them, going from `@sizeOf(DatePosix(i16)) == 4` to `@sizeOf(DatePosix(i32)) == 8` to `@sizeOf(DatePosix(i64)) == 16` is not trivial. It also changes the underlying `EpochDays` type.

I just typed these out as the three most common `Date` types I expect users to reach for:

- `Date16` is good for dates with years between -32_768 to 32_768 and suitable for most applications.
- `Date32` is good for dates with years between -2_147_483_648 to 2_147_483_648 and suitable for astronomy.
- `Date64` covers the age of the universe and is also suitable for astronomy.

On second thought, a single `Date` with `i16` is sufficient for the stdlib and most users. I'll refactor and add the comment with the supported range.

Reply:
Done. Let me know if you can think of common cases where `i16` isn't enough.